src/puppetlabs/comidi.clj

```diff
@@ -167,11 +167,13 @@
     (update-in route-info [:path] conj "!")
 
     (sequential? pattern)
-    (if-let [next (first pattern)]
-      (update-route-info*
-        (update-in route-info [:path] conj next)
-        (rest pattern))
-      route-info)
+    (loop [info route-info
+           [head & tail] pattern]
+        (if tail
+          (recur
+            (update-in info [:path] conj head)
+            tail)
```

Comment:
I think this is `concat` with different results depending of the type of sequence of `pattern`? But maybe not lazy? I don't know if one is faster than the other or we need the different behavior (from the schema I'd guess not), did we look at using one of the collection modifiers instead of a loop-recur?.

Reply:
Nope, that's a good idea to check out, this was a pretty naive refactor, just trying to keep it mostly the same

Reply:
The trick seems to be `into`, not concat, to get an array instead of a lazy seq.